#!/usr/bin/env python3
PACKAGE = "traffic_light_pkg"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("mode", int_t, 0, "0: Manual, 1: Crosswalk, 2: Intersection", 1, 0, 2)
gen.add("north_green_light_on", bool_t, 0, "North green light on (only for manual mode)", True)
gen.add("south_green_light_on", bool_t, 0, "South green light on (only for manual mode)", True)
gen.add("green_duration", int_t, 0, "Green duration in seconds", 45, 0, 60)
gen.add("red_duration", int_t, 0, "Red duration in seconds (overlap in intersection mode)", 15, 0, 60)

exit(gen.generate(PACKAGE, "traffic_light_pkg", "NonAdaptiveNS"))
